src/inmanta/util/__init__.py

```diff
@@ -863,3 +871,118 @@ def check_for_pool_exhaustion(self) -> None:
 
     def _reset_counter(self) -> None:
         self._exhausted_pool_events_count = 0
+
+
+def remove_comment_part(to_clean: str) -> str:
+    """
+    Remove the comment part of a given string and ensure that the length of the string is greater than 0
+
+    :param to_clean: The string to clean
+    :return: A cleaned string
+    """
+    # Refer to PEP 508. A requirement could contain a hashtag
+    to_clean = to_clean.strip()
+    if to_clean.startswith("#"):
+        raise ValueError("The name of the requirement cannot be a comment!")
+    drop_comment, _, _ = to_clean.partition(" #")
+    # We make sure whitespaces are not counted in the length of this string, e.g. "        #"
+    drop_comment = drop_comment.strip()
+    if len(drop_comment) == 0:
+        raise ValueError("The name of the requirement cannot be an empty string!")
+    return drop_comment
+
+
+CanonicalRequirement = typing.NewType("CanonicalRequirement", packaging.requirements.Requirement)
+
+
+def parse_requirement(requirement: str) -> CanonicalRequirement:
+    """
+    To be able to compare requirements, we need to make sure that every requirement's name is canonicalized otherwise issues
+    could arise when checking if packages are installed in a particular Venv.
+
+    :param requirement: The requirement's name
+    :return: A new requirement instance
+    """
+    # Packaging Requirement is not able to parse requirements with comment. Therefore, we need to remove the `comment` part
+    drop_comment = remove_comment_part(to_clean=requirement)
+    # We canonicalize the name of the requirement to be able to compare requirements and check if the requirement is
+    # already installed
+    # This instance is considered as doomed because the requirement name that this instance is using is not "canonicalized"
+    requirement_instance = packaging.requirements.Requirement(requirement_string=drop_comment)
+    requirement_instance.name = packaging.utils.canonicalize_name(requirement_instance.name)
+    canonical_requirement_instance = CanonicalRequirement(requirement_instance)
+    return canonical_requirement_instance
+
+
+def parse_requirements(requirements: Sequence[str]) -> list[CanonicalRequirement]:
```

Comment:
it should be documented if this method handles comments or not and how it handles them.